File: Examples/CSharp/AsposePdfFacades/Annotations/ExtractAnnotations.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;
using System;
using Aspose.Pdf.Annotations;
using System.Collections;

namespace Aspose.Pdf.Examples.CSharp.AsposePDFFacades.Annotations
{
    public class ExtractAnnotations
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_Annotations();
            //create PdfAnnotationEditor
            PdfAnnotationEditor annotationEditor = new PdfAnnotationEditor();
            //open PDF document
            annotationEditor.BindPdf(dataDir+ "ExtractAnnotations.pdf");
            //extract annotations
            Enum[] annotType = { AnnotationType.FreeText, AnnotationType.Line };
            ArrayList annotList = (ArrayList)annotationEditor.ExtractAnnotations(1, 2, annotType);
            for (int index = 0; index < annotList.Count; index++)
            {
                Annotation annotation = (Annotation)annotList[index];
                Console.WriteLine(annotation.Contents);
            } 
            
        }
    }
}

File: Examples/VisualBasic/AsposePdfFacades/Annotations/ExtractAnnotations.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades
Imports Aspose.Pdf.Annotations
Imports System.Collections
Namespace AsposePDFFacades.Annotations
    Public Class ExtractAnnotations
        Public Shared Sub Run()
            Try
                ' ExStart:ExtractAnnotations
                ' The path to the documents directory.
                Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_Annotations()
                ' Create PdfAnnotationEditor
                Dim annotationEditor As New PdfAnnotationEditor()
                ' Open PDF document
                annotationEditor.BindPdf(dataDir & Convert.ToString("ExtractAnnotations.pdf"))
                ' Extract annotations
                Dim annotType As [Enum]() = {AnnotationType.FreeText, AnnotationType.Line}
                Dim annotList As ArrayList = DirectCast(annotationEditor.ExtractAnnotations(1, 2, annotType), ArrayList)
                For index As Integer = 0 To annotList.Count - 1
                    Dim annotation As Annotation = DirectCast(annotList(index), Annotation)
                    Console.WriteLine(annotation.Contents)
                    ' ExEnd:ExtractAnnotations
                Next
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try
        End Sub
    End Class
End Namespace